### ISSUE TITLE
Advanced search options

### ISSUE THREAD

**Issue by josdejong** [josdejong/jsoneditor#43]
The current search in the JSON Editor searches for text matches within the fields and values of items. The search functionality can be expanded with options like:
- match case
- match words
- regex support
- filter on matching path or path parts
- match multiple results per field instead of a match of the fields as a whole

**Comment by josdejong** [comment]
This issue is very old already, I will close it now. Feel free to reopen it when there is need for it again.

**Comment by vikasgurjar** [comment]
Is advance search available with current version?

**Comment by josdejong** [comment]
That's not

**Comment by vikasgurjar** [comment]
With the Filter (truncate with JMESpath) option, can we have a way to search the required path and make changes to json, and when we do editor.get(), we get the whole json, instead of the truncated one(with the changes of-course). 

![image](https://user-images.githubusercontent.com/13435960/51251518-8ad25980-19bf-11e9-8391-fa8d6f8a12a7.png)

this gives the searched path. But it changes the state, so only the truncated json is available on editor.get()

**Comment by josdejong** [comment]
Yes indeed, this feature is to transform your JSON, it would be nice if you could use such a powerful query for just search too.
